pkg/kubectl/stop.go

```diff
@@ -241,6 +243,18 @@ func (reaper *JobReaper) Stop(namespace, name string, timeout time.Duration, gra
 	if err = scaler.Scale(namespace, name, 0, nil, retry, waitForJobs); err != nil {
 		return "", err
 	}
+	// at this point only dead pods are left, that should be removed
+	selector, _ := extensions.PodSelectorAsSelector(job.Spec.Selector)
+	podList, err := pods.List(selector, fields.Everything())
+	if err != nil {
+		return "", err
+	}
+	for _, pod := range podList.Items {
+		if err := pods.Delete(pod.Name, gracePeriod); err != nil {
+			return "", err
```

Comment:
This is also what rolling update does (https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/rolling_updater.go#L684). It's fine for now, but it would probably be better to try deleting all the pods and return all the errors using an aggregate error (https://github.com/kubernetes/kubernetes/blob/master/pkg/util/errors/errors.go).